bindings/java/src/main/native/jucx_common_def.cc

```diff
@@ -193,9 +193,9 @@ jobject process_request(void *request, jobject callback)
 
     // If request is a pointer set context callback and rkey.
     if (UCS_PTR_IS_PTR(request)) {
+        ucs_memory_cpu_store_fence();
```

Comment:
the memory barrier has to be right before setting jucx_request (and after setting the callback)
only the "flag" (aka ->jucx_request) should be after the write barrier

Reply:
Something like this?
```
struct jucx_context *fenced_request = ((struct jucx_context *)request;
ucs_memory_cpu_store_fence();
fenced_request->jucx_request = env->NewGlobalRef(jucx_request);
```